### DISCUSSION TITLE
关于img2pdf插件生成pdf的问题

### DISCUSSION THREAD

**Discussion by fordeslorn** [hect0x7/JMComic-Crawler-Python#258] [Q&A] [answered]
<img width="209" alt="0" src="https://github.com/user-attachments/assets/933afbb5-5599-4863-b190-1413422ac290">

如果一个本子有多个章节的话，使用img2pdf插件后，会生成散装的每个章节的pdf文件，该插件是否能生成一个按章节顺序的总pdf文件呢？
如果不行，我有两个ideas：
1）我搜索到pip install PyPDF2，该库似乎能合并pdf文件，注意到章节的顺序似乎是按album_id的大小排列的，基于此或许可以用排序算法保证按章节顺序合并，同时类似于压缩zip插件，可以提供在生成总pdf后删除原pdf文件的配置项
2）img2pdf插件如果在生成pdf时就可以按章节顺序生成总的pdf文件当然更好，省去了下载和删除原pdf的过程(但是似乎不好保证章节顺序)😦

**Comment by hect0x7** [comment]
可以生成按章节顺序的完整全本pdf。目前这个插件的代码只处理photo，你可以改造一下代码，接收album，遍历photo，调原方法处理photo就行了。

**Reply by fordeslorn** [reply]
不太会啊，还没学过py😢

**Reply by fordeslorn** [reply]
<img width="810" alt="111" src="https://github.com/user-attachments/assets/402594e1-75a7-4562-8351-157cfa57c0ac">
<img width="312" alt="626" src="https://github.com/user-attachments/assets/94022b7d-45f7-480e-bf72-1e51f5113fed">


大佬求助，在用这个代码下载303074的第23章时不知道为什么下载了两个一样的文件夹，现在这个文件夹不知道为什么无法移动和删除

**Reply by hect0x7** [reply]
看下op.yml的内容

**Reply by hect0x7** [reply]
> 现在这个文件夹不知道为什么无法移动和删除

你可以搜下windows强制删除文件夹命令

**Reply by fordeslorn** [reply]
<img width="719" alt="26526" src="https://github.com/user-attachments/assets/18080650-2bd0-4958-b3a9-b2c9164bd86c">

**Reply by hect0x7** [reply]
没看出什么问题，删除文件夹重新下载看看

**Reply by fordeslorn** [reply]
<img width="946" alt="11111111111111111111111" src="https://github.com/user-attachments/assets/2b33d1eb-087d-4500-9f7c-4363c07ca87c">
这两文件夹不知道为什么是互通的，我在一个文件夹新建了一个文件，另一个文件夹也出现了
<img width="325" alt="333333333333333333" src="https://github.com/user-attachments/assets/a7919829-cd75-4a9d-854c-cb6f8c3e2ddc">
删除显示这个，好像删不了

**Reply by fordeslorn** [reply]
> <img alt="11111111111111111111111" width="946" src="https://private-user-images.githubusercontent.com/159607131/352806502-2b33d1eb-087d-4500-9f7c-4363c07ca87c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNjc3NDYsIm5iZiI6MTcyMjE2NzQ0NiwicGF0aCI6Ii8xNTk2MDcxMzEvMzUyODA2NTAyLTJiMzNkMWViLTA4N2QtNDUwMC05ZjdjLTQzNjNjMDdjYTg3Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyOFQxMTUwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZjFiMzhlOTcxNzZjODEzOWRiNWQzZDAyMWI1NTcwNTMzNjZmZWU5YTBiZDM3NjJhOTY4YzhjNjYxMDJiODAwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.SCT2X8MWLlkdZt3iCWRenb2auYPugPia0dt-7ZYYDS8"> 这两文件夹不知道为什么是互通的，我在一个文件夹新建了一个文件，另一个文件夹也出现了 <img alt="333333333333333333" width="325" src="https://private-user-images.githubusercontent.com/159607131/352806507-a7919829-cd75-4a9d-854c-cb6f8c3e2ddc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNjc3NDYsIm5iZiI6MTcyMjE2NzQ0NiwicGF0aCI6Ii8xNTk2MDcxMzEvMzUyODA2NTA3LWE3OTE5ODI5LWNkNzUtNGE5ZC04NTRjLWNiNmY4YzNlMmRkYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyOFQxMTUwNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZmM0ZTdiOWU3ZmQ5NmM3NGE1MDc2MmQ5MDU5MDk4ZTE3ZWU5ODQ3MGI2NjRjMjlhYTk4NmVjNDVkMGI5NzViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.F2vXDBQegeHGYHu-TWndqcO_2g2F2MNj00Vqdzx9mMI"> 删除显示这个，好像删不了

解决了，用7zip压缩并删除原文件可以删掉，不知道这是什么bug

**Comment by hect0x7** [comment, marked as answer]
【img2pdf插件 支持合并整本】
我已添加实现代码：https://github.com/hect0x7/JMComic-Crawler-Python/pull/261
由于还有一些工作没有完成，暂时不会正式发这版的代码。你可以尝试把新插件的代码复制直接使用。

**Reply by fordeslorn** [reply]
感谢大佬😊
